### ISSUE TITLE
placeholder: for those having issues with six, kpm installation on el capitan osx

### ISSUE THREAD

**Issue by v1k0d3n** [kubespray/kpm-stackanetes#3]
this is just a placeholder (you can close as it's informational only):

if users have issues with installing `six` on el capitan, please do the following as a workaround to installing kpm.

```
sudo -H pip install kpm --upgrade --ignore-installed six
```

otherwise, users may experience the following issue (which is common):

``
Installing collected packages: Six, flask-cors, kpm
  Found existing installation: six 1.4.1
    DEPRECATION: Uninstalling a distutils installed project (Six) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
    Uninstalling six-1.4.1:
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg/pip/req/req_set.py", line 736, in install
    requirement.uninstall(auto_confirm=True)
  File "/Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg/pip/req/req_install.py", line 742, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 302, in move
    copy2(src, real_dst)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 131, in copy2
    copystat(src, dst)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 103, in copystat
    os.chflags(dst, st.st_flags)
OSError: [Errno 1] Operation not permitted: '/tmp/pip-R9cBAx-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/six-1.4.1-py2.7.egg-info'
``

just thought it may be helpful to track it here, even though the issue is *not* related to kpm.

**Comment by v1k0d3n** [comment]
closing issue; informational only. hopefully it saves 1+ users time searching for fixes elsewhere.

**Comment by ant31** [comment]
Thx, do you know how I can reproduce it without using your tip ?

**Comment by v1k0d3n** [comment]
kind of hard...i have a love/hate relationship with pip in general.

tl;dr version: use a vm in vmware fusion to test an initial installation of OS X with the project, or set up virtualenv with pyenv and pyenv-virtualenv. i'm in the process of documenting, and i'll place it up on my blog in a week or so. i'm just slammed atm. i found similar issues with the stackanetes project (for oslo.config), and running a clean virtualenv helped me catch and resolve that one as well.
